nomt/src/merkle/page_walker.rs

```diff
@@ -816,42 +789,50 @@ mod tests {
 
         let mut walker =
             PageWalker::<Blake3Hasher>::new(root, page_source, page_pool.clone(), None);
-        let mut write_pass = page_cache.new_write_pass();
 
         walker.advance_and_replace(
-            &mut write_pass,
             TriePosition::new(),
             vec![
                 (key_path![0, 1, 0, 1, 1, 0], val(1)),
                 (key_path![0, 1, 0, 1, 1, 1], val(2)),
             ],
         );
 
+        match walker.conclude() {
```

Comment:
this test was actually silently broken in master for a couple reasons.

1. it broke the contract of the page walker, because it called `advance_and_replace` with a position that was a sub-position of a previous `advance_and_replace` call.
2. it tested that there were 3 modified pages, however, in the previous version of the test there were only two pages that were being modified (root + 1 child page). the third page came from a broken invariant causing `build_stack` to reload a page which was already pushed to the `updated_pages` vector.

the changes in this PR exposed the problems with this test so I fixed them here.